scripts/license-headers

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+#
+# Iterates over all rust source files and ensures they start with the license
+# header as per the `.license-header` file at the root of the repository.
+
+set -euo pipefail
+IFS=$'\n'
+
+shopt -s globstar
+for file in */src/**/*.rs
```

Comment:
I just realised that this will miss out on tests and examples, so changed it to this:

```suggestion
shopt -s globstar extglob
for file in */+(src|tests|examples)/**/*.rs
```